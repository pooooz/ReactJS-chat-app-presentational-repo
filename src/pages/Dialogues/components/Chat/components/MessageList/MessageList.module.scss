.message__list {
  font-family: 'Press Start 2P', cursive;
  display: flex;
  flex-direction: column;
  padding: 0 100px 20px;
}