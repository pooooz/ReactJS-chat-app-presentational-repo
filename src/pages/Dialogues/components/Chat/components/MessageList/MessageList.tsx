import React, { FC } from 'react';
import { nanoid } from 'nanoid';

import { MessagePure, Message } from './components/Message/Message';

import styles from './MessageList.module.scss';

interface MessageListProps {
  messages: Message[];
}

export const MessageList: FC<MessageListProps> = (props) => {
  return (
    <div className={styles.message__list}>
      {props.messages.map((message) => (
        <MessagePure message={message} key={nanoid()} />
      ))}
    </div>
  );
};

export const MessageListPure = React.memo(MessageList);
