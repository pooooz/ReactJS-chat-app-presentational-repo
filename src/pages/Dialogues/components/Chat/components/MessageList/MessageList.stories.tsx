import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { MessageList } from 'src/pages/Dialogues/components/Chat/components/MessageList/MessageList';
import { WithClasses } from 'src/HOC/WithClasses';
import styles from '../../Chat.module.scss';

export default {
  title: 'pages/Dialogues/components/MessageList',
  component: MessageList,
} as ComponentMeta<typeof MessageList>;

const messages = [
  { id: '1', author: 'Admin', text: 'Initialization...' },
  { id: '2', author: 'You', text: 'Hello!!!' },
];

export const Default: ComponentStory<typeof MessageList> = (args) => (
  <MessageList messages={args.messages} />
);
Default.args = {
  messages,
};

const MessageListWithClasses = WithClasses(MessageList);
export const Bordered: ComponentStory<typeof MessageList> = (args) => (
  <MessageListWithClasses {...args} classes={styles.border} />
);
Bordered.args = {
  messages,
};
