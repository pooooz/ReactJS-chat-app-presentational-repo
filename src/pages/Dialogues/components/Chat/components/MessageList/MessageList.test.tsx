import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { MessageList } from './MessageList';

describe('MessageList', () => {
  it('Render component', () => {
    render(
      <MessageList messages={[{ id: '1', author: 'Render', text: 'Render' }]} />
    );
  });

  it('Render with snapshot', () => {
    const { asFragment } = render(
      <MessageList messages={[{ id: '1', author: 'Render', text: 'Render' }]} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('Render with text', () => {
    render(
      <MessageList messages={[{ id: '1', author: 'Render', text: 'Render' }]} />
    );
    expect(screen.getByText(/Author: <\w*>/));
  });
});
