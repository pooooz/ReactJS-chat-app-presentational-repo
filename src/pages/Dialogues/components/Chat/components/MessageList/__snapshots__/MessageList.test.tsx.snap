// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList Render with snapshot 1`] = `
<DocumentFragment>
  <div
    class="message__list"
  >
    <div
      class="message"
    >
      <p
        class="glitch"
        data-text="Render"
      >
        Render
      </p>
      <p
        class="glitch"
        data-text="Author: <Render>"
      >
        Author: &lt;Render&gt;
      </p>
    </div>
  </div>
</DocumentFragment>
`;
